### PR TITLE
Parsing and validation working as discussed

### PR DESCRIPTION
**Pull request by PauliusMinialga**
Modified map_tester a bit and left my comments there so it would be clear what is happening and why. Put everything in a new folder for easier testing and checking 

Left my comment in parser and validation sections as well so it would be clear why I made those choices. Tested everything with the maps we had and it seems to be working, if you have any remarks and comments please let me know, so I can fix things.

I am unable to run MLX42 on my mac (will try to solve the issue, cause it's annoying), so for now I won't be able to test everything together

### CODE REVIEW

**Review by vovashko — COMMENTED** [review]
thanks! I have a few comments before we merge:
 - in the header file, I see that you commented out some of the duplicate functions(not all), however, that prevented the program to be built with flags. Let's make sure to build with flags, so that we can eliminate errors early. It's not that big but still had to clean up and comment out some things before I could make a tester build.
 - for reference this is how I build it without the makefile, we can use it for unit testing: 
 _cc -Wall -Wextra -Werror parsing_validation_done/parser.c parsing_validation_done/validator.c main.c gnl/get_next_line.c -o testing_parser -I. libft/libft/libft.a_

- great you've added the echo statements in the shell script, it is easier to detect in which map the problem occurred. I'm not quite sure about your comment on check_format. if you were referring to the check_file_content function that has commented part in which we check if the files for textures can be opened, that wouldn't catch a duplicate texture for example. So if you can take a look if this is being caught somewhere else, that would be swell. 

TLDR clean up duplicate references from the header and remove unnecessary comments if applicable so the build can run smoothly. Would be good if you add your testing main to a separate file, so we don't need to go back and forth commenting things out. 

PS. don't hesitate to remove the files/functions from the folder/header in which we separated parsing and validation functions before. if we don't need them they would have to go regardless. it will only makes things cleaner

**Comment by PauliusMinialga** [comment]
Did some changes according to your comments:
1 - rewrote some parts of the code and used the TEST_MODE 2 to have the init_game and init_mfd functions connected to my code for easier testing
2 - deleted the folder we had created for parsing and validation, as well deleted the map_validation and map_parser files, cause they are not needed anymore
3 - added a check for duplicate textures and colours
4 - added a main_parser.c where I put my main function and used this line to test all those files (cc -Wall -Wextra -Werror parsing_validation_done/parser.c parsing_validation_done/validator.c init.c main_parser.c gnl/get_next_line.c -o testing_parser -I. libft/libft/libft.a)
5 - for now I want to leave the code as it is. If you check it and it works for you, then I'll start getting rid of all for loops and making it norminette approved

**Review by vovashko — APPROVED** [review]
Thanks! All worked fine.  Merging at this stage
